server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailSetMethod.scala

```diff
@@ -202,19 +202,33 @@ case class UpdateResults(results: Seq[UpdateResult]) {
 
   private def doUpdate(messageId: MessageId, update: EmailSetUpdate, storedMetaData: List[ComposedMessageIdWithMetaData], session: MailboxSession): SMono[UpdateResult] = {
     val mailboxIds: List[MailboxId] = storedMetaData.map(metaData => metaData.getComposedMessageId.getMailboxId)
-    resetFlags(messageId, update, mailboxIds, session)
+
+    val reset = resetFlags(messageId, update, mailboxIds, session)
       .onErrorResume(e => SMono.just[UpdateResult](UpdateFailure(EmailSet.asUnparsed(messageId), e)))
       .switchIfEmpty(SMono.just[UpdateResult](UpdateSuccess(messageId)))
+
+    SFlux.concat(reset, updateMailboxIds(messageId, update, session))
+      .reduce(UpdateSuccess(messageId), (u1, u2) => {
```

Comment:
can we find better names than u1 and u2 ? 

(at least when you un-wip this :-) )